File: python/day-1/a.py
def find_product(nums):
    for n1 in nums:
        for n2 in nums:
            if n1 + n2 == 2020:
                return n1 * n2

def main():
    nums = [int(s) for s in open('input.txt').read().split()]
    print(find_product(nums))

main()

File: rust/day-1/a.rs
use std::fs;
use std::error::Error;

fn main() -> Result<(), Box<dyn Error>> {
    let inp = fs::read_to_string("input.txt")?;

    let nums: Vec<i32> = inp
        .lines()
        .map(|line| line.parse().unwrap())
        .collect();

    println!("{}", find_product(nums));
    Ok(())
}
    
    
fn find_product(nums: Vec<i32>) -> i32 {
    for s1 in &nums {
        for s2 in &nums {
            if s1 + s2 == 2020 {
                return s1 * s2;
            }
        }
    }
    0
}